waltz-service/src/main/java/com/khartec/waltz/service/involvement/InvolvementService.java

```diff
@@ -133,4 +167,45 @@ private Involvement mkInvolvement(EntityReference entityReference,
                 "waltz");
     }
 
+
+    private void logChange(EntityReference entityReference, String userId, EntityInvolvementChangeCommand command) {
+        String message = String.format("Involvement kind (%s) %s for person: %s",
+                resolvePrettyInvolvementKind(command.involvementKindId()),
+                command.operation().name().toLowerCase(),
+                resolveNames(command.personEntityRef()));
+
+        ImmutableChangeLog changeLog = ImmutableChangeLog.builder()
+                .parentReference(entityReference)
+                .message(message)
+                .userId(userId)
+                .childKind(command.personEntityRef().kind())
+                .operation(command.operation())
+                .build();
+        changeLogService.write(changeLog);
+    }
+
+
+    private String resolvePrettyInvolvementKind(long id) {
+        if(involvementKindIdToNameMap == null) {
+            this.involvementKindIdToNameMap = loadInvolvementKindIdToNameMap();
+        }
+
+        return String.format("[%s [%s]]", this.involvementKindIdToNameMap.get(id), id);
+    }
+
+
+    private List<String> resolveNames(EntityReference... refs) {
```

Comment:
Don't understand why we need to use a list - will we ever have more than one name to resolve... ?

Reply:
good point, corrected